otlp/common.go

```diff
@@ -110,6 +110,75 @@ func IsClassicApiKey(key string) bool {
 	return false
 }
 
+func IsLegacyAPIKey(key string) bool {
+	keyLen := len(key)
+
+	if keyLen == 32 {
+		// Check if all characters are hex digits (0-9, a-f)
+		for i := 0; i < keyLen; i++ {
+			c := key[i]
+			if !((c >= '0' && c <= '9') || (c >= 'a' && c <= 'f')) {
+				return false
+			}
+		}
+		return true
+	} else if keyLen == 64 {
+		// Check the prefix pattern "hc[a-z]ic_"
+		if key[:2] != "hc" || key[3:6] != "ic_" {
+			return false
+		}
+		if key[2] < 'a' || key[2] > 'z' {
+			return false
+		}
+
+		// Check if the remaining characters are alphanumeric lowercase
+		for i := 6; i < keyLen; i++ {
+			c := key[i]
+			if !((c >= '0' && c <= '9') || (c >= 'a' && c <= 'z')) {
+				return false
+			}
+		}
+		return true
+	}
+
+	return false
+}
+
+func IsLegacyAPIKeySwitched(key string) bool {
```

Comment:
I recommend this implementation replace `IsClassicApiKey()` and the two other implementations be removed, then merge.

Reply:
Agreed